runtime/common/src/traits.rs

```diff
@@ -135,6 +135,14 @@ pub trait Leaser {
 
 	/// Returns the current lease period.
 	fn lease_period_index() -> Self::LeasePeriod;
+
+	/// Returns true if the parachain already has a lease in any of lease periods in the inclusive
+	/// range `[first_period, last_period]`.
+	fn already_leased(
```

Comment:
Thinking out loud a bit here, this could also fail if an auction ever covered any lease periods that are in the past because we simply would not be able to check for intersection. AFAICT this is not an issue because in `do_new_auction` we check

```
ensure!(lease_period_index >= T::Leaser::lease_period_index(), Error::<T>::LeasePeriodInPast);
```

but just wanted to point out in case there is some other method of creating an auction that may be used in the future. (But also why we would create an auction that covers something in the past?).

Reply:
yeah we should not concern ourselves with auctions in the past as they are not relevant for the present or future :)